src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceEducation.vue

```diff
@@ -0,0 +1,261 @@
+          ></v-checkbox>
+        </form>
+      </v-col>
+    </div>
+  </v-row>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+
+import EducationList, { type EducationData } from "@/components/EducationList.vue";
+import { useAlertStore } from "@/store/alert";
+import type { EceEducationProps } from "@/types/input";
+import type { Components } from "@/types/openapi";
+import * as Rules from "@/utils/formRules";
+export default defineComponent({
+  name: "EceEdducation",
+  components: { EducationList },
+  props: {
+    props: {
+      type: Object as () => EceEducationProps,
+      required: true,
+    },
+    modelValue: {
+      type: Object as () => { [id: string]: Components.Schemas.Transcript },
+      required: true,
+    },
+  },
+  emits: {
+    "update:model-value": (_educationData: { [id: string]: Components.Schemas.Transcript }) => true,
+  },
+  setup: () => {
+    const alertStore = useAlertStore();
+
+    return {
+      alertStore,
+    };
+  },
+  data: function () {
+    return {
+      clientId: "",
+      mode: "add",
+      id: "",
+      previousSchool: "",
+      school: "",
+      program: "",
+      campusLocation: "",
+      studentName: "",
+      studentNumber: "",
+      language: "",
+      startYear: "",
+      endYear: "",
+      officialTranscriptRequested: false,
+      officialTranscriptReceived: false,
+      Rules,
+    };
+  },
+  computed: {
+    localId() {
+      return Object.keys(this.modelValue).length + 1;
+    },
+  },
+  mounted() {
+    if (Object.keys(this.modelValue).length === 0) {
+      this.mode = "add";
+    } else {
+      this.mode = "list";
+    }
+  },
+  methods: {
+    async handleSubmit() {
+      // Validate the form
+      const { valid } = await (this.$refs.addEducationForm as any).validate();
+
+      if (valid) {
+        // Prepare the new or updated Transcript data
+        const newTranscript: Components.Schemas.Transcript = {
+          id: null,
+          educationalInstitutionName: this.school,
+          programName: this.program,
+          campusLocation: this.campusLocation,
+          studentName: this.studentName,
+          studentNumber: this.studentNumber,
+          languageofInstruction: this.language,
+          startDate: this.startYear,
+          endDate: this.endYear,
+        };
+        const clientId = this.clientId ? this.clientId : this.localId;
+        // Update the modelValue dictionary
+        const updatedModelValue = {
+          ...this.modelValue,
+          [clientId]: newTranscript,
+        };
+
+        // Emit the updated modelValue
+        this.$emit("update:model-value", updatedModelValue);
+
+        // Set success alert message
+        const message = this.modelValue[clientId] ? "You have successfully edited your Education." : "You have successfully added your Education.";
```

Comment:
I'm a bit confused on the difference between localId and clientId. 

Should this be educationId?

Reply:
I think it's to determine which education to remove in the future for two cases.

1) we grabbed the education that was saved to the DB 
2) we just added a new education from the DB. 

I feel like we could just refer to the index in this instance.

Reply:
We likely need to modify the names of some properties. 'localId' refers to the ID of a new transcript (just in FrontEnd) as we can not set a value for transcript.Id because we are sending that data to server, while 'clientId' is used when editing an already existing transcript.